drivers/wallclock/sun6i-cnt64.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <error.h>
+#include <mmio.h>
+#include <drivers/clock.h>
+#include <drivers/wallclock.h>
+#include <util.h>
+
+#define  OSC24M_CNT64_CTRL_REG		0x28
+#define OSC24M_CNT64_LOW_REG		0x284
+#define OSC24M_CNT64_HIGH__REG		0x288
+
+static uint64_t
+sunxi_wallclock_get_time(struct device *dev)
+{
+	uint64_t time;
+	uint32_t high_reg;
+	uint32_t low_reg
+
+	mmio_setbits32(dev->regs + OSC24M_CNT64_CTRL_REG, BIT(1);
+
+	while(mmio_read32(dev->regs + OSC24M_CNT64_CTRL_REG) ){
+
+		low_reg = mmio_read32(dev->regs + OSC24M_CNT64_LOW_REG);
+		high_reg =  mmio_read32(dev->regs + OSC24M_CNT64_HIGH_REG);
+		time = ((uint64_t)low_reg << 32) | high_reg
+	}
+
+	return time;
+}
+
+static int
+sunxi_wallclock_probe(struct device *devi __unused)
+{
+	uint32_t Counter_Latch = mmio_read32(dev->regs + PSC24M_CNT64_CTRL_REG);
```

Comment:
This function runs once at boot. Please explain what you're trying to do here.

Reply:
that ensures the counter is not latched at start up as it should not be. I guess i should have commented my code

Reply:
You can't "un-latch" the counter. You have to latch it every time you read it (so the high and low words are consistent with each other). It stays latched at one value until you set the bit again, and then it updates to the new value.